.github/workflows/nightly-fips.yml

```diff
@@ -0,0 +1,107 @@
+name: Nightly FIPS
```

Comment:
Would be nice to "templatize" some of these files. Not sure would the best option would be, but something to consider.

Pre-generation hooks, jsonnet, etc may all be good options. Not blocking this PR though

Reply:
Hmm, does GitHub Actions natively support those options? If not, we'd have to use something like a pre-commit hook. `matrix` strategies in GitHub actions get use decently far